predicate/predicate.py

```diff
@@ -174,4 +164,85 @@ def _regex(self, lookup_model, lookup_field):
     def _iregex(self, lookup_model, lookup_field):
         return bool(re.search(self.value, lookup_field, flags=re.I))
 
+    def _in(self, instance, lookup_value):
+        """
+        Evaluates whether the lookup list and this value list intersect.
+        """
+        return bool(set(lookup_value) & set(self.value))
+
+    def _any(self, instance, lookup_value):
+        """
+        Evaluates whether the lookup list contains this value.
+        """
+        return self.value in lookup_value
```

Comment:
I think this should be the intended semantics of `_exact`, and constitutes a bug. A good invariant to expect of the `P` class would be that `model in P(**kw)` if and only if `MyModel.objects.filter(Q(**kw), pk=model.pk).exists()`.  

In the case of relations where you'd get  back a list of instances (`ManyToManyField` and a reverse foreign key relation), the exact lookup would get evaluated to a SQL join that would *only* include matching rows. I.E. I think `MyModel.objects.filter(m2m_relation__attribute='something')` would match the intended semantics of `_any`. 

I'll try to make a test case reproducing this.